apollo/lib/checkWinCondition.tsx

```diff
@@ -254,6 +375,14 @@ export default function checkWinConditions(
     return null;
   }
 
+  const optionalObjectiveAmount = checkOptionalObjectiveAmountWinCondition(
+    map,
+    actionResponse,
+  );
+  if (optionalObjectiveAmount) {
+    return optionalObjectiveAmount;
+  }
```

Comment:
This is kind of a "hot-loop", especially when the AI runs. Before calling the function, this should verify that the `ActionResponse` is either `OptionalObjective` or `DeniedOptionalObjective` to avoid calling the `checkOptionalObjectiveAmountWinCondition` unless necessary.

Reply:
done in [81a2e0e](https://github.com/nkzw-tech/athena-crisis/pull/47/commits/81a2e0ea2540b5d9056d57e0abf1ceade69e1a0c)